streaming/src/main/java/org/apache/carbondata/streaming/segment/StreamSegment.java

```diff
@@ -400,12 +386,7 @@ public static void recoverFileIfRequired(
   public static CarbonFile[] listDataFiles(String segmentDir) {
     CarbonFile carbonDir = FileFactory.getCarbonFile(segmentDir);
     if (carbonDir.exists()) {
-      return carbonDir.listFiles(new CarbonFileFilter() {
-        @Override
-        public boolean accept(CarbonFile file) {
-          return CarbonTablePath.isCarbonDataFile(file.getName());
-        }
-      });
+      return carbonDir.listFiles(file -> CarbonTablePath.isCarbonDataFile(file.getName()));
```

Comment:
Are we target on jdk8 from now on? If yes, codes in `org.apache.carbondata.common` like `Maps`, `Strings` can be removed

Reply:
For carbon 2.0, the answer is yes, in my opinion.

we already use new featrues of jdk8(lambda, stream API)

Reply:
OK. 

FYI: I check [spark release note](https://spark.apache.org/docs/2.2.0/), support for Java 7 was removed since 2.2